### PR TITLE
[2/X] Add servlet for retrieving signups by opportunity id.

### PR DESCRIPTION
**Pull request by nikhitashanker**
- Add servlet OpportunitySignupDataServlet for retrieving signups by opportunity id
- Add tests for servlet

### CODE REVIEW

**Comment by nikhitashanker** [comment]
Presubmit errors will be fixed by events PRs.

File: project/src/test/java/com/google/sps/OpportunitySignupDataTest.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        new VolunteeringOpportunity.Builder(EVENT_ID, NAME, NUMBER_OF_SPOTS).build();
+    SpannerTasks.insertVolunteeringOpportunity(opportunity);
+    HttpServletRequest request = Mockito.mock(HttpServletRequest.class);
+    HttpServletResponse response = Mockito.mock(HttpServletResponse.class);
+    StringWriter stringWriter = new StringWriter();
+    PrintWriter writer = new PrintWriter(stringWriter);
+    Mockito.when(request.getParameter(OPPORTUNITY_ID)).thenReturn(opportunity.getOpportunityId());
+    Mockito.when(response.getWriter()).thenReturn(writer);
+
+    new OpportunitySignupDataServlet().doGet(request, response);
```
Comment (yicyao) [inline project/src/test/java/com/google/sps/OpportunitySignupDataTest.java:63]: I think you can do in setup() like
opportunitySignupServlet = new OpportunitySignupDataServlet();
based on https://stackoverflow.com/a/5465522

File: project/src/test/java/com/google/sps/OpportunitySignupDataTest.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+import org.mockito.Mockito;
+import org.springframework.mock.web.MockServletContext;
+
+
+/** Test that tests getting the signups for a volunteering opportunity. */
+@RunWith(JUnit4.class)
+public final class OpportunitySignupDataTest {
+  private static final String NAME = "Performer";
+  private static final int NUMBER_OF_SPOTS = 240;
+  private static final String OPPORTUNITY_ID = "opportunity-id";
```
Comment (dash102) [inline project/src/test/java/com/google/sps/OpportunitySignupDataTest.java:33]: nit: Name this to specify that is it a parameter rather than the actual ID

Comment (nikhitashanker) [reply project/src/test/java/com/google/sps/OpportunitySignupDataTest.java:33]: Done - agreed, this adds clarity.

File: project/src/test/java/com/google/sps/OpportunitySignupDataTest.java (hunk 1 above)
Comment (nikhitashanker) [reply project/src/test/java/com/google/sps/OpportunitySignupDataTest.java:63]: Done.

**Pull request merged**